extensions/standard-processors/processors/ListenSyslog.cpp

```diff
@@ -187,7 +187,7 @@ void ListenSyslog::runThread() {
         struct sockaddr_in cli_addr;
         clilen = sizeof(cli_addr);
         int recvlen = recvfrom(_serverSocket, _buffer, sizeof(_buffer), 0, (struct sockaddr *) &cli_addr, &clilen);
-        if (recvlen > 0 && (uint64_t) (recvlen + getEventQueueByteSize()) <= _recvBufSize) {
+        if (recvlen > 0 && (uint64_t) (recvlen + getEventQueueByteSize()) <= gsl::narrow<uint64_t>(_recvBufSize)) {
```

Comment:
Unsafe narrowing. `_recvBufSize` is parsed from a property, meaning it has to be validated. `gsl::narrow` should be used when you "promise" the value will fit in the destination type, or otherwise you're willing to crash the program.

Reply:
Updated in 267cda83a77a5a6d05f9df714d0c923814351826